Bottom: 75d7c7fd613caa688d7ae1e29bd1d4845566cf49
Top:    14a5ab25884a69bd56d49c2a9771d87aa2de932e
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 14:04:14 -0500

Add buildbot NTEL worker Docker image

---
diff --git a/ansible-playbook/group_vars/all/buildbot.yml b/ansible-playbook/group_vars/all/buildbot.yml
index 6b069548..d5a1a447 100644
--- a/ansible-playbook/group_vars/all/buildbot.yml
+++ b/ansible-playbook/group_vars/all/buildbot.yml
@@ -1,3 +1,4 @@
 ---
 buildbot_version: 1.1.1
 buildbot_worker_port: 9989
+buildbot_worker_ntel: worker-ntel
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index 7c9bb550..3fa1e99f 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -18,3 +18,4 @@ buildsystem_database_root_password:  '{{ lookup("password", inventory_dir + "/cr
 buildsystem_database_build_password: '{{ lookup("password", inventory_dir + "/credentials/buildsystem.novatech-llc.com/database/readuser") }}'
 svn_username:                        '{{ lookup("file",     inventory_dir + "/personal_credentials/svn/username") }}'
 svn_password:                        '{{ lookup("file",     inventory_dir + "/personal_credentials/svn/password") }}'
+buildbot_worker_ntel_password:       '{{ lookup("password", inventory_dir + "/credentials/buildbot.novatech-llc.com/worker_ntel length=20") }}'
diff --git a/roles/buildbot-worker-ntel/README.md b/roles/buildbot-worker-ntel/README.md
new file mode 100644
index 00000000..225dd44b
--- /dev/null
+++ b/roles/buildbot-worker-ntel/README.md
@@ -0,0 +1,38 @@
+Role Name
+=========
+
+A brief description of the role goes here.
+
+Requirements
+------------
+
+Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance, if the role uses the EC2 module, it may be a good idea to mention in this section that the boto package is required.
+
+Role Variables
+--------------
+
+A description of the settable variables for this role should go here, including any variables that are in defaults/main.yml, vars/main.yml, and any variables that can/should be set via parameters to the role. Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
+
+Dependencies
+------------
+
+A list of other roles hosted on Galaxy should go here, plus any details in regards to parameters that may need to be set for other roles, or variables that are used from other roles.
+
+Example Playbook
+----------------
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+
+    - hosts: servers
+      roles:
+         - { role: username.rolename, x: 42 }
+
+License
+-------
+
+BSD
+
+Author Information
+------------------
+
+An optional section for the role authors to include contact information, or a website (HTML is not allowed).
diff --git a/roles/buildbot-worker-ntel/defaults/main.yml b/roles/buildbot-worker-ntel/defaults/main.yml
new file mode 100644
index 00000000..e2a4a407
--- /dev/null
+++ b/roles/buildbot-worker-ntel/defaults/main.yml
@@ -0,0 +1,23 @@
+---
+# defaults file for buildbot-worker-ntel
+
+# The hostname passed as an envirnment variable into the container
+container_addr_map:
+  buildbot:
+    ip_addr: '127.0.0.1'
+    hostname: buildbot-worker-ntel.example.com
+
+# Network names
+docker_network_frontend: 'frontend'
+
+buildbot_worker_data_path: '/buildbot'
+
+# the name of the image being duilt and used for the container
+buildbot_image_repo: https://github.com/novatechweb/docker-buildbot-worker-ntel.git
+buildbot_image_name: '{{ docker_registry_username }}/buildbot-worker-ntel'
+
+# the name of the container being started
+buildbot_container_name: '{{ docker_name_prefix }}buildbot-worker-ntel'
+
+# the name of the data-volume used by the container
+buildbot_dv_name: '{{ docker_name_prefix }}buildbot-worker-ntel_DV'
diff --git a/roles/buildbot-worker-ntel/handlers/main.yml b/roles/buildbot-worker-ntel/handlers/main.yml
new file mode 100644
index 00000000..6347b70a
--- /dev/null
+++ b/roles/buildbot-worker-ntel/handlers/main.yml
@@ -0,0 +1,2 @@
+---
+# handlers file for buildbot-worker-ntel
\ No newline at end of file
diff --git a/roles/buildbot-worker-ntel/meta/main.yml b/roles/buildbot-worker-ntel/meta/main.yml
new file mode 100644
index 00000000..72237997
--- /dev/null
+++ b/roles/buildbot-worker-ntel/meta/main.yml
@@ -0,0 +1,57 @@
+galaxy_info:
+  author: your name
+  description: your description
+  company: your company (optional)
+
+  # If the issue tracker for your role is not on github, uncomment the
+  # next line and provide a value
+  # issue_tracker_url: http://example.com/issue/tracker
+
+  # Some suggested licenses:
+  # - BSD (default)
+  # - MIT
+  # - GPLv2
+  # - GPLv3
+  # - Apache
+  # - CC-BY
+  license: license (GPLv2, CC-BY, etc)
+
+  min_ansible_version: 1.2
+
+  # If this a Container Enabled role, provide the minimum Ansible Container version.
+  # min_ansible_container_version:
+
+  # Optionally specify the branch Galaxy will use when accessing the GitHub
+  # repo for this role. During role install, if no tags are available,
+  # Galaxy will use this branch. During import Galaxy will access files on
+  # this branch. If Travis integration is configured, only notifications for this
+  # branch will be accepted. Otherwise, in all cases, the repo's default branch
+  # (usually master) will be used.
+  #github_branch:
+
+  #
+  # platforms is a list of platforms, and each platform has a name and a list of versions.
+  #
+  # platforms:
+  # - name: Fedora
+  #   versions:
+  #   - all
+  #   - 25
+  # - name: SomePlatform
+  #   versions:
+  #   - all
+  #   - 1.0
+  #   - 7
+  #   - 99.99
+
+  galaxy_tags: []
+    # List tags for your role here, one per line. A tag is a keyword that describes
+    # and categorizes the role. Users find roles by searching for tags. Be sure to
+    # remove the '[]' above, if you add tags to this list.
+    #
+    # NOTE: A tag is limited to a single word comprised of alphanumeric characters.
+    #       Maximum 20 tags per role.
+
+dependencies: []
+  # List your role dependencies here, one per line. Be sure to remove the '[]' above,
+  # if you add dependencies to this list.
\ No newline at end of file
diff --git a/roles/buildbot-worker-ntel/tasks/bootstrap.sh b/roles/buildbot-worker-ntel/tasks/bootstrap.sh
new file mode 100755
index 00000000..8353ad9d
--- /dev/null
+++ b/roles/buildbot-worker-ntel/tasks/bootstrap.sh
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+DIRS="/cache \
+      /cache/ccache \
+      /cache/downloads \
+      /cache/images \
+      /cache/ipkg-repository \
+      /cache/mirrors \
+      /cache/releases \
+      /cache/sstate"
+
+for d in $DIRS
+do
+  mkdir -p $d
+  chown -c -R buildbot:buildbot $d
+  chmod -v 777 $d
+done
diff --git a/roles/buildbot-worker-ntel/tasks/main.yml b/roles/buildbot-worker-ntel/tasks/main.yml
new file mode 100644
index 00000000..5d3d8e3f
--- /dev/null
+++ b/roles/buildbot-worker-ntel/tasks/main.yml
@@ -0,0 +1,2 @@
+---
+# tasks file for buildbot-worker-ntel
\ No newline at end of file
diff --git a/roles/buildbot-worker-ntel/tests/inventory b/roles/buildbot-worker-ntel/tests/inventory
new file mode 100644
index 00000000..878877b0
--- /dev/null
+++ b/roles/buildbot-worker-ntel/tests/inventory
@@ -0,0 +1,2 @@
+localhost
+
diff --git a/roles/buildbot-worker-ntel/tests/test.yml b/roles/buildbot-worker-ntel/tests/test.yml
new file mode 100644
index 00000000..98df706d
--- /dev/null
+++ b/roles/buildbot-worker-ntel/tests/test.yml
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - buildbot-worker-ntel
\ No newline at end of file
diff --git a/roles/buildbot-worker-ntel/vars/main.yml b/roles/buildbot-worker-ntel/vars/main.yml
new file mode 100644
index 00000000..afdab38e
--- /dev/null
+++ b/roles/buildbot-worker-ntel/vars/main.yml
@@ -0,0 +1,2 @@
+---
+# vars file for buildbot-worker-ntel
\ No newline at end of file
